### ISSUE TITLE
Opening Two-Factor Authentication settings causes two duplicate API requests for EnableTwoFactorAuth

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#48034]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.24-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @tgolen 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1724704755405909

## Action Performed:
1. Go to staging.new.expensify.com
2. Click Account settings > Security > Two Factor authentication setting
## Expected Result:
No duplicate calls API calls
## Actual Result:
Two duplicate API requests for EnableTwoFactorAuth
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/3743ae8c-483d-4d12-b03e-37032fff6aec

![Full Screen #9](https://github.com/user-attachments/assets/435705be-a429-4135-a77b-9d9bc3fc3584)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by tgolen** [comment]
Thanks for the investigation @Nodebrute. I think the thing that I really dislike about that is now dev is a very different environment than production, which makes testing and development much more difficult. We are being forced into some weird patterns (using a bunch of `useRef`) to try and get development to work more like we expect production to work.

Do you know if React is planning to make production work in this concurrent mode like dev is running in? I guess that would return it to an apples-to-apples situation at some point, which is better, but I really don't like having to guess "does this only happen in dev or not".

I'm going to assign this to external and get a C+ to review your proposal to implement (because even though it only happens in dev, it messed me up enough that I think it's worth fixing).

**Comment by tgolen** [comment]
@mollfpr Are you OK with the proposal [here](https://github.com/Expensify/App/issues/48034#issuecomment-2311238332)? I think it's OK to move forward with.

**Comment by johncschuster** [comment]
@mollfpr bump on the above?

**Comment by mollfpr** [comment]
Sorry for the delay!

>Are you OK with the proposal https://github.com/Expensify/App/issues/48034#issuecomment-2311238332? I think it's OK to move forward with.

@tgolen The issue seems to affect other pages for example `OpenReport` and `OpenInitialSettingsPage `. I'm onboard if we decide to specifically fix the issue for this 2FA page.

Also, the solution doesn't fix your issue here https://github.com/Expensify/App/pull/48030#issuecomment-2321849748.

**Comment by tgolen** [comment]
> Also, the solution doesn't fix your issue here https://github.com/Expensify/App/pull/48030#issuecomment-2321849748.

OK, thanks!

> @tgolen The issue seems to affect other pages for example OpenReport and OpenInitialSettingsPage .

> @mollfpr The issue affects any API calls made within useEffect because Strict Mode runs effects twice. To ensure consistency between production and development, we can turn off Strict Mode.

OK, that is a bit unfortunate. Maybe we should discuss this on Slack in the open-source channel?

**Comment by tgolen** [comment]
I started a conversation about it in Slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1725376236233599).

**Comment by johncschuster** [comment]
Thanks for doing that, @tgolen! 🙌

**Comment by johncschuster** [comment]
It looks like the discussion in Slack may still be ongoing.

**Comment by johncschuster** [comment]
@mollfpr I haven't been able to follow the discussion very closely. Do you feel we've come to a conclusion? If so, can you summarize where we've landed?

**Comment by mollfpr** [comment]
>I haven't been able to follow the discussion very closely. Do you feel we've come to a conclusion? If so, can you summarize where we've landed?

We haven't had any conclusion yet, but we might have a holistic solution to fix this issue. I think it's better to move this issue weekly/monthly and I'll keep visiting the thread.

**Comment by tgolen** [comment]
I think there was enough support internally for disabling strict mode by default until there is a more clear problem statement and project detailing what is expected to be gained from having strict mode enabled.

@Nodebrute Can you please update your proposal and I'll go ahead and assign this to you.

**Comment by johncschuster** [comment]
Waiting on QA to test the fix. @tgolen, it looks like Applause is asking on the PR which URL should be used for staging. Can you let them know?

**Comment by mollfpr** [comment]
I have replied to the question.

**Comment by mollfpr** [comment]
@johncschuster I think we can use the date for this issue https://github.com/Expensify/App/issues/48791 closed. It's 11th September.

**Comment by johncschuster** [comment]
# Payment Summary:

Contributor: @Nodebrute paid $250 via [Upwork](https://www.upwork.com/nx/wm/offer/103846483) - PAID

Contributor+: @mollfpr due $250 via NewDot

Upwork job [here](https://www.upwork.com/nx/wm/offer/103846483)! Please apply

@mollfpr can you please provide regression test steps if they are necessary?

**Comment by mollfpr** [comment]
@johncschuster I think we don't need a regression step since it's only affecting the development side and it's knowingly a feature from React but we happen not to need it yet.

**Comment by johncschuster** [comment]
Ok great! Thanks for clarifying that! Please request payment!

**Comment by garrettmknight** [comment]
$250 approved for @mollfpr
